src/main/java/com.amazon.bookstore/BookstoreController.java

```diff
@@ -3,21 +3,48 @@
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.beans.factory.annotation.Autowired;
+
 
 @Controller
 public class BookstoreController {
 
+    @Autowired
+    BookRepo bookRepo;
+
     @GetMapping("/")
     public String index(Model model) {
-        Book b1 = new Book("Harry Potter", "JK Rowling", "N/A", 123, "HP", "Fantasy", 1);
-        Book b2 = new Book("Lord of the Rings", "JRR Tolkien", "N/A", 124, "LOTR", "Fantasy", 2);
         BookStore books = new BookStore();
-        books.addBook(b1);
-        books.addBook(b2);
+        for(Book b : bookRepo.findAll()){
+            books.addBook(b);
+        }
+        model.addAttribute("bookstore", books);
+
+        return "bookstore";
+    }
 
+    @PostMapping("/bookstore")
+    public String mainpage(Model model) {
+        BookStore books = new BookStore();
+        for(Book b : bookRepo.findAll()){
+            books.addBook(b);
+        }
         model.addAttribute("bookstore", books);
 
         return "bookstore";
     }
 
+    @PostMapping("/editBookstore")
+    public String uploadBookstore(){
+        //functionality here
```

Comment:
What functionality? Is it missing?

Reply:
This is for Lynn's ticket. I just added this so all the modifications and additions the owner can do is done in one page rather than multiple pages.

Reply:
Oh perfect. Thanks for clarifying.